src/composite/chain.jl

```diff
@@ -9,10 +9,11 @@ export ChainBlock, chain
 user defined blocks horizontically. It is a `Vector`
 like composite type.
 """
-struct ChainBlock{N, T, MT <: AbstractBlock{N, T}} <: CompositeBlock{N, T}
-    blocks::Vector{MT}
+struct ChainBlock{N, T} <: CompositeBlock{N, T}
+    blocks::Vector{AbstractBlock{N, T}}
```

Comment:
This cause the following won't work

```julia
list = []
for k in 1:3
   push!(list, X)
end

ChainBlock(list)
```

It would better to have a check and conversion completely dynamically. or just roll back this change.

Reply:
I think this is ok, we should use `chain` interface to handle this.